pkg/runcfanotify/runcfanotify.go

```diff
@@ -161,7 +166,7 @@ func cmdlineFromPid(pid int) []string {
 // generates an event on the notifier. This is automatically called for new
 // containers detected by RuncNotifier, but it can also be called for
 // containers detected externally such as initial containers.
-func (n *RuncNotifier) AddWatchContainerTermination(containerID string, containerPID int) error {
+func (n *RuncNotifier) AddWatchContainerTermination(containerID string, containerPID int, bundleDir string) error {
```

Comment:
It is used below lines 183 and 191

Reply:
Yes, but `watchContainerTermination` and `watchContainerTerminationFallback` only use it to generate the `ContainerEvent`, i.e, it's only passed to make sure the events have this field set, so my point is, should we provide this field on `EventTypeRemoveContainer` events too? It seems to me it's something the caller could save and if needed later on could be retrieved using the ContainerID.

Reply:
Ah, I see your point. It is similar with containerID: it is mostly used to fill the events. Well, it is also used to avoid redundant calls but that could be done with the containerPID.

Let me think how the callers can do it. There are two callers:
1. In pkg/container-collection/options.go WithRuncFanotify() but we can't pass the bundleDir there anyway.
2. In pkg/runcfanotify/runcfanotify.go, but in a different context. We could have the struct RuncNotifier to keep track of bundleDirs for each container, but that's more complex. Do you think it is worth it?

Reply:
Let's go back one step: is bundledir needed in the remove event? Do we want to provide all container data on that event too or is it a task of the caller to maintain it if needed later on?

Reply:
If we want to emulate OCI hooks, we need the bundle directory and all other container data in the remove event too (in runc, the [OCI State](https://github.com/opencontainers/runtime-spec/blob/main/runtime.md#state) is passed to the [Poststop hook](https://github.com/opencontainers/runtime-spec/blob/main/config.md#poststop)).

If we don't do that, the caller could maintain the container data for each container ID or PID, but that's potentially more work for the caller.

However, as opposed to the prestart hook, the poststop hook as emulated by the runcfanotify pkg is not synchronous: it does not guarantee that the directory still exists at the time the hook is executed. So the hook might not be able to inspect the files there...

Reply:
I am ok with removing Bundle in the remove event, and revisit when we have a more concrete use case. Wdyt?

Reply:
@mauriciovasquezbernal Updated.